### PR TITLE
Feature/form functionality

### PR DESCRIPTION
**Pull request by knees4bees**
## **_PR TEMPLATE_**
1. Is this a fix or a feature?
* Feature.
* 
2. What is the change?
* Category icons and borders now change colors when clicked; tiny foundation for number validation.
* 
3. What does it fix?
* N/A.
* 
4. Where should the reviewer start?
* Start with reviewing the JS file: do the variables and functions make sense? Are they tied to the relevant elements in the DOM? Some changes were made to the HTML and CSS files to accommodate the JS functionality. Also see comment on line 18 of `main.js` about a function that's ripe for refactoring. Any ideas?
* 
5. How should this be tested?
* Check that clicking on the activity boxes has the desired effect.

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -19,16 +19,19 @@ <h2 class="subhead">New Activity</h2>
           <article class="categories">
             <h3 class="form-label">Select a category:</h3>
             <section class="category-box-wrapper">
-              <div class="category-boxes">
-                <img class="category-image" src="./assets/study.svg" alt="light bulb above an open book">
+              <div class="category-boxes study-box">
+                <img class="category-image study-image" src="./assets/study.svg" alt="light bulb above an open book">
+                <img class="category-image study-image-active hidden" src="./assets/study-active.svg" alt="green light bulb above an open book">
```
Comment (stevecalla) [inline index.html:24]: Consider adding alt text = light bulb above an open book selected by user.

File: index.html [hunk 2]
```diff
@@ -19,16 +19,19 @@ <h2 class="subhead">New Activity</h2>
...
                 <p class="category-text">Study</p>
               </div>
-              <div class="category-boxes">
-                <img class="category-image" src="./assets/meditate.svg" alt="lotus flower">
+              <div class="category-boxes meditate-box">
+                <img class="category-image meditate-image" src="./assets/meditate.svg" alt="lotus flower">
+                <img class="category-image meditate-image-active hidden" src="./assets/meditate-active.svg" alt="purple lotus flower">
```
Comment (stevecalla) [inline index.html:29]: Consider adding alt text = lotus flower selected by user.

File: index.html [hunk 3]
```diff
@@ -19,16 +19,19 @@ <h2 class="subhead">New Activity</h2>
...
                 <p class="category-text">Meditate</p>
               </div>
-              <div class="category-boxes">
-                <img class="category-image b" src="./assets/exercise.svg" alt="sneaker">
+              <div class="category-boxes exercise-box">
+                <img class="category-image exercise-image" src="./assets/exercise.svg" alt="sneaker">
+                <img class="category-image exercise-image-active hidden" src="./assets/exercise-active.svg" alt="red sneaker">
```
Comment (stevecalla) [inline index.html:34]: Consider adding alt text =  exercise selected by user.

File: main.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+var secondsInput = document.querySelector('.seconds-input');
+
+
+
+categoryBoxWrapper.addEventListener('click', activateCategory);
+
+// TODO: this function is ripe for refactoring but I'm not sure how yet
+function activateCategory(event) {
+  if (event.target.classList.contains('study-box')) {
+    studyImage.classList.toggle('hidden');
```
Comment (stevecalla) [inline main.js:21]: Let's create a function for this hide/add and call here.

File: main.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    studyImageActive.classList.toggle('hidden');
+    studyBox.classList.toggle('study-border');
+  } else if (event.target.classList.contains('meditate-box')) {
+    meditateImage.classList.toggle('hidden');
```
Comment (stevecalla) [inline main.js:25]: Let's create a function for this hide/add and call here as a starter to go SRP.

File: main.js [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    meditateImageActive.classList.toggle('hidden');
+    meditateBox.classList.toggle('meditate-border');
+  } else if (event.target.classList.contains('exercise-box')) {
+    exerciseImage.classList.toggle('hidden');
```
Comment (stevecalla) [inline main.js:29]: Let's create a function for this hide/add and call here.

File: main.js [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+
+
+
+categoryBoxWrapper.addEventListener('click', activateCategory);
+
+// TODO: this function is ripe for refactoring but I'm not sure how yet
+function activateCategory(event) {
+  if (event.target.classList.contains('study-box')) {
+    studyImage.classList.toggle('hidden');
+    studyImageActive.classList.toggle('hidden');
```
Comment (stevecalla) [inline main.js:22]: Nice use of toggle.

File: main.js [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+    studyBox.classList.toggle('study-border');
+  } else if (event.target.classList.contains('meditate-box')) {
```
Comment (stevecalla) [inline main.js:24]: good research on this topic!

**Review by stevecalla — APPROVED** [review]

File: main.js (hunk 6 above)
Comment (stevecalla) [reply main.js:29]: I have the code for this; also need to modify based on the functionality

File: main.js (hunk 5 above)
Comment (stevecalla) [reply main.js:25]: I have the code for this; also need to modify based on the functionality

File: main.js (hunk 4 above)
Comment (stevecalla) [reply main.js:21]: I have the code for this; also need to modify based on the functionality

File: index.html (hunk 1 above)
Comment (knees4bees) [reply index.html:24]: yep, let's put this on the list

File: main.js (hunk 4 above)
Comment (knees4bees) [reply main.js:21]: cool

File: main.js (hunk 7 above)
Comment (knees4bees) [reply main.js:22]: 😃

**Review by knees4bees — COMMENTED** [review]
addressed @stevecalla 's comments; let's implement in the next code push

**Pull request merged**